### PR TITLE
chore(state): remove Full online pruner

### PR DESCRIPTION
**Pull request by qdm12**
## Changes

Remove the (buggy, incomplete, unused) full online pruner, in order to unblock #3005 

## Issues

Related to #1973 and #2835

### CODE REVIEW

**Comment by jimjbrettj** [comment]
Do we have plans to store the current pruner in a different branch or something before deletion? That way we can reference in the future

**Review by jimjbrettj — APPROVED** [review]
Looks good, just left a comment

**Comment by qdm12** [comment]
> Do we have plans to store the current pruner in a different branch or something before deletion? That way we can reference in the future

The new pruner is being worked on at #2831 (blocked by #3050) which I'll rebase once this gets merged, and will probably finish it sometime in February during my "off time" 😸 We don't really need the older online pruner code since it's really buggy/flawed, and the new one is already separate/quite different (still flawed regarding fork-awareness as you pointed out 👍)

File: cmd/gossamer/flags.go
Comment (EclesioMeloJunior) [inline cmd/gossamer/flags.go:371]:
```suggestion
		Usage: `State trie online pruning ("archive")`,
```
```suggestion
		Usage: `State trie online pruning (""archive")`,
```
```suggestion
		Usage: `State trie online pruning (""archive")`,
```

File: dot/state/service_integration_test.go
Comment (EclesioMeloJunior) [inline dot/state/service_integration_test.go:182]: is this related to https://github.com/ChainSafe/gossamer/pull/2831?

**Review by EclesioMeloJunior — APPROVED** [review]

File: dot/state/service_integration_test.go
Comment (qdm12) [reply dot/state/service_integration_test.go:182]: Yes since `pruner.Full` got removed. This test is to be unskipped and this line uncommented (modified) in #2381). I added a comment on the other PR to make sure I don't forget 😉

**Pull request merged**
